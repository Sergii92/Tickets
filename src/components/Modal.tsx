import React from "react";
import { ToastContainer, toast, Zoom } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { useSelector, useDispatch } from "react-redux";
import { Modal, Button } from "react-bootstrap";
import { Col, Row } from "react-bootstrap";

import { Сhair } from "../styled/pages";

import { dataActions, selectPlaces } from "../redux/ducks/session";
import { IplaceData } from "../interfaces";

const MyModal = ({ show, handleClose, sesssion }: any) => {
  const dispatch = useDispatch();
  const notify = () => {
    toast.success("Booled Success !", {
      position: "top-center",
      autoClose: 4000,
      transition: Zoom,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,
    });
  };
  const placesData: Array<IplaceData> = useSelector(selectPlaces);

  const freePlaces: Array<IplaceData> = placesData
    .filter((places: IplaceData) => places.sessionID === sesssion.id)
    .filter((place: IplaceData) => place.booked === false);

  const selectchair = (id: number) => dispatch(dataActions.bookedPlace(id));

  const putPlacesData = (data: Array<IplaceData>) => {
    dispatch(dataActions.putPlaces(data));

    notify();
    setTimeout(() => {
      handleClose();
    }, 4000);
  };

  return (
    <Modal show={show} onHide={handleClose}>
      <Modal.Header closeButton>
        <Modal.Title>{sesssion.sessionName}</Modal.Title>
      </Modal.Header>
      <Modal.Header>
        <Modal.Title>Время Сеанса:{sesssion.time}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{sesssion.description}</Modal.Body>
      <Modal.Footer style={{ justifyContent: "center" }}>
        <Row>
          {placesData &&
            placesData
              .filter((places) => places.sessionID === sesssion.id)
              .map((place) => {
                return (
                  <Сhair
                    key={place.id}
                    onClick={() => selectchair(place.id)}
                    booked={place.booked}
                  />
                );
              })}
        </Row>
        <Row>
          <Col>
            <Button
              variant="primary"
              size="lg"
              onClick={() => putPlacesData(placesData)}
            >
              Reserve
            </Button>
            <ToastContainer
              position="top-center"
              autoClose={4000}
              hideProgressBar={false}
              newestOnTop={false}
              transition={Zoom}
              closeOnClick
              rtl={false}
              pauseOnFocusLoss
              draggable
            />
          </Col>
          <Col>Свободных мест:{freePlaces.length}</Col>
        </Row>
      </Modal.Footer>
    </Modal>
  );
};

export default MyModal;
