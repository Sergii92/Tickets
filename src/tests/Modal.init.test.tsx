import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Modal from "../components/Modal";
import { First } from "react-bootstrap/lib/Pagination";

const mockStore = configureStore();

describe("Day component render", () => {
  const props = {
    show: true,
    handleClose: () => {},
    sesssion: {
      daysId: 1,
      time: "10:00",
      id: 11,
      description:
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam possimus quaerat ipsa dolor nihil cum ullam iste culpa veniam, magni a neque beatae consequatur totam eaque maxime aperiam illum molestiae.",
      sessionName: "Sunday session1",
    },
  };

  const store = mockStore(props);
  it("RENDER", () => {
    const component = shallow(
      <Provider store={store}>
        <Modal {...props} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  // it("////", () => {
  //   const component = shallow(
  //     <Provider store={store}>
  //       <Modal {...props} />
  //     </Provider>
  //   );
  //   // expect(component.find(".modal-title").first().text()).toEqual(
  //   //   "Sunday session1"
  //   // );
  // });
});
