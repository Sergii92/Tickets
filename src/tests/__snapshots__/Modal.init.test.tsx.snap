// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component render RENDER 1`] = `ShallowWrapper {}`;

exports[`Day component render get Modal snapshot 1`] = `ShallowWrapper {}`;
